### PR TITLE
[AKS] Bugfix/sl 2062/notes display incosistency

### PR DESCRIPTION
**Pull request by leszekgu**
Mostly cleanup.

### CODE REVIEW

File: src/app/hearing-part/components/hearings-table/hearings-table.component.ts [hunk 1]
```diff
@@ -78,11 +80,16 @@ export class HearingsTableComponent implements AfterViewChecked {
     }
 
     hasNotes(hearing: HearingForListingWithNotes): boolean {
-        return hearing.notes.length > 0;
+        return hearing.notes.filter(note => {
+            return note.type === NoteType.OTHER_NOTE ||
```
Comment (yarpo) [inline src/app/hearing-part/components/hearings-table/hearings-table.component.ts]: We have a helper method `isOfTypeOtherOrListing`. Maybe we should add something similar here?

This probably has some business meaning, such as: "isNoteAllowedTo..."

File: src/app/hearing-part/components/listing-note-list/listing-note-list.component.ts [hunk 2]
```diff
@@ -1,73 +1,68 @@
-import { ListingCreateNotesConfiguration } from './../../models/listing-create-notes-configuration.model';
+import { ListingCreateNotesConfiguration } from '../../models/listing-create-notes-configuration.model';
 import { Component, Input, ViewChild } from '@angular/core';
 import { NoteListComponent } from '../../../notes/components/notes-list/note-list.component';
-import { Note } from '../../../notes/models/note.model';
-import { NoteViewmodel, getNoteViewModel } from '../../../notes/models/note.viewmodel';
+import { isOfTypeOtherOrListing, Note } from '../../../notes/models/note.model';
+import { getNoteViewModel, NoteViewmodel, sortNotesByLatestFirst } from '../../../notes/models/note.viewmodel';
```
Comment (yarpo) [inline src/app/hearing-part/components/listing-note-list/listing-note-list.component.ts]: I know it's very minor and it was done automatically by IDE, but I'd suggest to import class first and static methods later.

File: src/app/hearing-part/components/listing-note-list/listing-note-list.component.ts [hunk 3]
```diff
@@ -1,73 +1,68 @@
...
+        this.oldNoteViewModels = sortNotesByLatestFirst(this.oldNoteViewModels);
+    }
 
-  prepareNotes() {
-    const preparedFreeTextNotes = this.notePreparerService.prepare(
-        this.specialNotesList.getModifiedNotes(),
-        this.entityId,
-        this.listingNotesConfig.entityName
-    );
+    constructor(readonly listingNotesConfig: ListingCreateNotesConfiguration,
```
Comment (yarpo) [inline src/app/hearing-part/components/listing-note-list/listing-note-list.component.ts]: is `listingNotesConfig` public by default? If so - why do we define `public` for the other argument?

I'd rather use private/public here even if redundant (but I'm not going to argue about it - personal preference).

File: src/app/notes/models/note.model.ts [hunk 4]
```diff
@@ -23,6 +24,10 @@ export function getNoteFromViewModel(note: NoteViewmodel): Note {
     } as Note
 }
 
+export function isOfTypeOtherOrListing(note: Note) {
```
Comment (yarpo) [inline src/app/notes/models/note.model.ts]: unit tests?

Yes, it's trivial logic, but still logic.

File: src/app/hearing-part/components/hearings-table/hearings-table.component.ts [hunk 5]
```diff
@@ -136,7 +143,7 @@ export class HearingsTableComponent implements AfterViewChecked {
             sortDirection: this.sort.direction
         };
 
-        this.tableSettingsSource$.next(ts)
+        this.tableSettingsSource$.next(ts);
```
[the remark this answers is not in the document]
Comment (r-kruk-kainos) [reply src/app/hearing-part/components/hearings-table/hearings-table.component.ts]: You can always git commit --allow-empty @leszekgu

File: src/app/hearing-part/components/hearings-table/hearings-table.component.spec.ts [hunk 6]
```diff
@@ -5,6 +5,8 @@ import * as moment from 'moment';
 import { Priority, priorityValue } from '../../models/priority-model';
 import { HearingsTableComponent } from './hearings-table.component';
 import { DEFAULT_HEARING_FOR_LISTING_WITH_NOTES, HearingForListingWithNotes } from '../../models/hearing-for-listing-with-notes.model';
+import { Note } from '../../../notes/models/note.model';
+import { NoteType } from '../../../notes/models/note-type';
```
[the remark this answers is not in the document]
Comment (leszekgu) [reply src/app/hearing-part/components/hearings-table/hearings-table.component.spec.ts]: model tends to be response from server, viewmodel a model used by the application (gathers data from few responses), type is a new one but I can see it's for an enum.

File: src/app/hearing-part/components/listing-note-list/listing-note-list.component.ts (hunk 3 above)
Comment (leszekgu) [reply src/app/hearing-part/components/listing-note-list/listing-note-list.component.ts]: good question, I would even follow it, why both of them are public, if not used from html.
I can change it to private as it should be.

File: src/app/notes/models/note.model.ts (hunk 4 above)
Comment (leszekgu) [reply src/app/notes/models/note.model.ts]: 80% test coverage, so yes this is too trivial logic imho.

File: src/app/hearing-part/components/hearings-table/hearings-table.component.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (leszekgu) [reply src/app/hearing-part/components/hearings-table/hearings-table.component.ts]: thx

File: src/app/hearing-part/components/hearings-table/hearings-table.component.spec.ts [hunk 7]
```diff
@@ -33,12 +35,36 @@ describe('HearingsTableComponent', () => {
...
     describe('Component ', () => {
-        it('has notes should properly verify notes of hearingparts', () => {
+        it('hasNotes should properly verify notes of hearingparts', () => {
             let hasNotes = component.hasNotes(generateHearings('asd'));
 
             expect(hasNotes).toBeFalsy();
         });
 
+        it('hasNotes should return true when hearing contains listing note type', () => {
+            let hearingWithNotes = { ...DEFAULT_HEARING_FOR_LISTING_WITH_NOTES,
```
[the remark this answers is not in the document]
Comment (leszekgu) [reply src/app/hearing-part/components/hearings-table/hearings-table.component.spec.ts]: fixed

File: src/app/hearing-part/components/hearings-table/hearings-table.component.ts (hunk 1 above)
Comment (leszekgu) [reply src/app/hearing-part/components/hearings-table/hearings-table.component.ts]: changed

**Pull request merged**